config.py

```diff
@@ -1,84 +1,123 @@
 import numpy as np
-from os import environ, path
+from os import environ, path, cpu_count
 from dotenv import load_dotenv
+import torch
 
+# Load environment variables from .env file
 BASE_DIR = path.abspath(path.join(path.dirname(__file__), "."))
 load_dotenv(path.join(BASE_DIR, ".env"))
 
-GPU_SIZE_PER_WORKER = float(environ.get("GPU_SIZE_PER_WORKER", 0.2))
-STORAGE_GPU_SIZE = float(environ.get("STORAGE_GPU_SIZE", 0.1))
-BUFFER_GPU_SIZE = float(environ.get("BUFFER_GPU_SIZE", 0.02))
-TRAINER_GPU_SIZE = float(environ.get("TRAINER_GPU_SIZE", 0.2))
 
-# IMPORTANT: Change this value to the number of cpu cores you want to use (recommended 80% of cpu)
-NUM_CPUS = int(environ.get("NUM_CPUS", 14))
-
-DEVICE = str(environ.get("DEVICE", "cuda"))
-IMITATION = eval(environ.get("IMITATION", False))
-CHAMP_DECIDER = eval(environ.get("CHAMP_DECIDER", False))
+def get_bool_env(var_name, default="False"):
+    """Converts environment variable to a boolean."""
+    return environ.get(var_name, default).lower() in ("true", "1", "t")
+
+
+def get_float_env(var_name, default=0.0):
+    """Fetches an environment variable and converts it to float."""
+    return float(environ.get(var_name, default))
+
+
+def get_int_env(var_name, default=0):
+    """Fetches an environment variable and converts it to int."""
+    return int(environ.get(var_name, default))
+
+def get_num_cpus(percentage=0.8):
+    """Calculates the number of CPU cores to use based on a percentage of the total available cores.
+    
+    Args:
+        percentage (float): The percentage of total cores to use (80% by default).
+
+    Returns:
+        int: The calculated number of cores to use, or the value from NUM_CPUS environment variable if set.
+    """
+    max_cpus = cpu_count() or 1  # Use at least 1 core if os.cpu_count() returns None
+    default_cores = max(1, int(max_cpus * percentage))  # Calculate default, ensure at least 1
+    
+    # Fetch NUM_CPUS from env, falling back to calculated default_cores if not specified
+    num_cpus = get_int_env("NUM_CPUS", default_cores)
+    
+    # Ensure NUM_CPUS does not exceed the actual number of available cores
+    return min(num_cpus, max_cpus)
+
+# IMPORTANT: Change this value to the percentage of CPU cores you want to use (default 80%)
+NUM_CPUS = NUM_CPUS = get_num_cpus()
+
+# Dynamically set the device based on available hardware acceleration
+if torch.cuda.is_available():

```

Comment:
This is going to become a big else if statement eventually. Is there a way to clean this up a bit and say basically, which device does your system support. Does torch have something like this?